File: src/desktop/ui/src/Analysis/Models/ReadingArea.ts
import type { ReadingShape } from "./ReadingShape";
import type { ReadingParticipant } from "./ReadingParticipant";
import type { ReadingRelationship } from "./ReadingRelationship";
import type { ReadingStatement } from "./ReadingStatement";

export interface ReadingArea {
  readonly id: string;
  readonly title: string;
  readonly summary: ReadingStatement | null;
  readonly shape: ReadingShape;
  readonly participants: readonly ReadingParticipant[];
  readonly relationships: readonly ReadingRelationship[];
  readonly orderedSteps: readonly ReadingStatement[];
  readonly purposes: readonly ReadingStatement[];
}

File: src/desktop/src-tauri/src/analysis/models/reading_area.rs
use crate::analysis::models::{
    ReadingParticipant, ReadingRelationship, ReadingShape, ReadingStatement,
};
use serde::{Deserialize, Serialize};

/// Represents one reading area, a bounded narrative region of the change.
#[derive(Clone, Debug, Deserialize, PartialEq, Eq, Serialize)]
#[serde(rename_all = "camelCase", deny_unknown_fields)]
pub struct ReadingArea {
    pub id: String,
    pub title: String,
    pub summary: Option<ReadingStatement>,
    pub shape: ReadingShape,
    pub participants: Vec<ReadingParticipant>,
    pub relationships: Vec<ReadingRelationship>,
    pub ordered_steps: Vec<ReadingStatement>,
    pub purposes: Vec<ReadingStatement>,
}
